test/controllers/passwords_controller_test.rb

```diff
@@ -29,4 +29,24 @@ def put_update_with_params
     put_update_with_params
     assert_redirected_to custom_path
   end
+
+  test 'redirect to http://test.host/users/sign_in after successfully sent reset password instructions' do
+    Devise::PasswordsController.any_instance.stubs(:successfully_sent?).with(@user).returns(true)
+
+    post :create, params: { "user" => { "email" => @user.email } }
+    assert_redirected_to 'http://test.host/users/sign_in'
+  end
+
+  test 'redirect to http://test.host/users/password/new after unsuccessfully sent reset password instructions' do
+    Devise::PasswordsController.any_instance.stubs(:successfully_sent?).with(@user).returns(false)
```

Comment:
There is any change we can set up this test scenario without stubbing the `successfully_sent?` method? We are stubbing an internal method of our controller, and with the stubbing we are not testing it's implementation (which is a core piece of the changes from this Pull Request).